dev-docs/rfcs/0002-persistent-stream-flag.md

```diff
@@ -0,0 +1,125 @@
+- Start Date: 2019-03-07
+- RFC PR: [#403](https://github.com/uber/xviz/pull/403)
+- XVIZ Issue: [#64](https://github.com/uber/xviz/issues/64)
+
+# Summary
+
+Propose the addition of a new state_update type **PERSISTENT** that will treat the data as being
+persistent to allow send-once data to be preserved for the duration of a session.
+
+In addition to the new update type we are adding an additional field to the stream_set,
+`no_data_streams`, that will be an explicit statement that this listed streams contain no data.
+
+# Motivation
+
+The majority of XVIZ data is time-based but the need to send and display data that is persistent
+occurs frequently with map data.
+
+Today we only have one way to control the valid duration of stream data, which is the XVIZ Config
+setting `TIME_WINDOW`.
+
+Streetscape.gl, and any XVIZ compliant viewer, can get the XVIZ Config `TIME_WINDOW` which can be
+set by the application to set the window of time that the viewer should inspect to find a stream
+datum. The first datum found should be the active datum for the current viewing time. This is an
+application wide mechanism so does not allow for individual stream control.
+
+This control is application wide and not stream specific, therefore we needed to add something at
+the data level to enable persistent data.
+
+In addition to adding persistent data, we want a way to invalidate persistent data in some way to
+provide some level of state management from the XVIZ data source.
+
+# Proposal
+
+In order to mark a stream as containing persistent data that should not be purged for the duration
+of a session I propose the following.
+
+### 1. Add a new update_type **PERSISTENT** to the **state_update** message
+
+Any primitive data included in this state update should remain visible outside the TIME_WINDOW
+setting.
+
+Only **primitives** are currently defined for the **PERSISTENT** message. Other XVIZ data types
+behavior is undefined with regard to this RFC. We will address them at a later date.
+
+Specific features for a state_update message with the update_type **PERSISTENT**:
+
+- A stream's data will be rendered regardless of the timestamp
```

Comment:
I disagree with this, I think timing should work the same as normal streams minus `TIME_WINDOW`.  Without that consideration how does `no_data_streams` work?  Also when you scrub through with the playback control which data should you show?

Reply:
Ah I see.  and your right.  I described a *GLOBAL* msg rather than a *PERSISTENT* message here